File: src/main/java/me/testaccount666/serversystem/listener/executables/prefixchat/ChatVaultAPI.java
package me.testaccount666.serversystem.listener.executables.prefixchat;

import lombok.Getter;
import org.bukkit.Bukkit;

public class ChatVaultAPI {
    @Getter
    private net.milkbowl.vault.chat.Chat _chat;

    public static boolean isVaultInstalled() {
        return Bukkit.getServer().getPluginManager().getPlugin("Vault") != null;
    }

    public boolean setupChat() {
        if (!isVaultInstalled()) return false;

        var registeredProvider = Bukkit.getServer().getServicesManager().getRegistration(net.milkbowl.vault.chat.Chat.class);
        if (registeredProvider == null) return false;

        _chat = registeredProvider.getProvider();
        return true;
    }
}

File: src/main/kotlin/me/testaccount666/serversystem/listener/executables/chat/prefixchat/ChatVaultAPI.kt
package me.testaccount666.serversystem.listener.executables.chat.prefixchat

import net.milkbowl.vault.chat.Chat
import org.bukkit.Bukkit

class ChatVaultAPI {
    var chat: Chat? = null
        private set

    fun setupChat(): Boolean {
        if (!isVaultInstalled) return false

        val registeredProvider = Bukkit.getServer().servicesManager.getRegistration(Chat::class.java) ?: return false

        chat = registeredProvider.getProvider()
        return true
    }

    companion object {
        val isVaultInstalled by lazy {
            return@lazy Bukkit.getServer().pluginManager.getPlugin("Vault") != null
        }
    }
}
